### DISCUSSION TITLE
Animation on unmount ?

### DISCUSSION THREAD

**Discussion by bricejar** [pmndrs/react-spring#1094] [Support]
Is is possible to animate a component on unmount ?

**Comment by bricejar** [comment]
Maybe by using the return function of useEffect ?

**Reply by aleclarson** [reply]
By that time, it's too late. You'll want to use the `useTransition` hook:
https://www.react-spring.io/docs/hooks/use-transition

**Reply by bricejar** [reply]
But what is the useTransition hook itself is unmounted ?

**Reply by aleclarson** [reply]
The component using `useTransition` is the parent of what's being animated, added, or removed. So the answer to your question is: "Make sure it doesn't or the animation will be interrupted by unmount"

**Reply by CapablancaYEHA** [reply]
@aleclarson 
Joining the conversation to get some help with `trail,unique, reset` properties for useTransition.
What if i dont see any visual difference adding them, but i`m pretty sure that provided items are mounted\unmouted.
I just dont get enter\leave animations for items changing, it all fires only on first items mount, not their change\update

**Reply by bricejar** [reply]
> 
> 
> The component using `useTransition` is the parent of what's being animated, added, or removed. So the answer to your question is: "Make sure it doesn't or the animation will be interrupted by unmount"

Well sometimes it is impossible to "make sure" so I guess this is a limitation of the curring React rendering model that cannot "pause" an unmount so as to perform an animation.

**Reply by aleclarson** [reply]
@bricejar Correct

@CapablancaYEHA You'll want to open an issue for that, with a sandbox plz 👍
